### PR TITLE
Reimplemented help screen easter eggs

### LINKED ISSUE #1280: Reimplement help screen easter egg messages
Accidentally got rid of it while adding the help screen, that's not intended and should be reimplemented.

### LINKED ISSUE #1300: Make the question mark button in the microbe stage open the help menu
The question mark `?` button in the microbe GUI is currently disabled. It should be made enabled and it should be made to basically open the pause menu and go to the help section there.

### PR DESCRIPTION
**Pull request by Kasterisk**
I also made the help texts and easter egg messages read from a json file. And implemented the functionality of the help button next to the suicide button.

Closes #1300, closes #1280 

@hhyyrylainen Seems like this pull request will cause a merge conflict with your latest branch. (sorry 'bout that)

### CODE REVIEW

File: src/microbe_stage/MicrobeHUD.cs [hunk 1]
```diff
@@ -242,7 +242,9 @@ public override void _Ready()
         populationLabel = GetNode<Label>(PopulationLabelPath);
         patchLabel = GetNode<Label>(PatchLabelPath);
         editorButton = GetNode<TextureButton>(EditorButtonPath);
-        helpScreen = GetNode<Control>(HelpScreenPath);
+        helpScreen = GetNode<HelpScreen>(HelpScreenPath);
+
+        helpScreen.BuildHelpTexts("MicrobeStage");
```
Comment (hhyyrylainen) [inline src/microbe_stage/MicrobeHUD.cs:247]: I think that making this an editor configurable property would be nice. And then load the help texts in the help screen _Ready method.

File: src/microbe_stage/editor/MicrobeEditorGUI.cs [hunk 2]
```diff
@@ -335,10 +335,12 @@ public override void _Ready()
         patchAmmoniaSituation = GetNode<TextureRect>(PatchAmmoniaSituationPath);
         patchPhosphateSituation = GetNode<TextureRect>(PatchPhosphateSituationPath);
         rigiditySlider = GetNode<Slider>(RigiditySliderPath);
-        helpScreen = GetNode<Control>(HelpScreenPath);
+        helpScreen = GetNode<HelpScreen>(HelpScreenPath);
 
         mapDrawer.OnSelectedPatchChanged = (drawer) => { UpdateShownPatchDetails(); };
 
+        helpScreen.BuildHelpTexts("MicrobeEditor");
```
Comment (hhyyrylainen) [inline src/microbe_stage/editor/MicrobeEditorGUI.cs:342]: same here

File: src/microbe_stage/editor/MicrobeEditorGUI.cs [hunk 3]
```diff
@@ -545,20 +547,12 @@ internal void OnNewCellClicked()
         editor.CreateNewMicrobe();
     }
 
-    internal void ToggleHelpScreen()
+    internal void OpenHelpScreen()
```
Comment (hhyyrylainen) [inline src/microbe_stage/editor/MicrobeEditorGUI.cs:550]: I think this and that other toggle method renaming is not the best, as this method actually toggles the help screen, but always hides the menu...

File: src/microbe_stage/MicrobeHUD.cs [hunk 4]
```diff
@@ -473,8 +476,12 @@ public void ToggleHelpScreen()
 
         if (!helpScreen.Visible)
         {
-            helpScreen.Show();
+            if (!GetTree().Paused)
+                GetTree().Paused = true;
```
Comment (hhyyrylainen) [inline src/microbe_stage/MicrobeHUD.cs:480]: Why is this if here? Why not just directly set paused to true?

Comment (Kasterisk) [reply src/microbe_stage/MicrobeHUD.cs:480]: I wasn't even aware it's there, I'll remove it

**Comment by hhyyrylainen** [comment]
The close button in the editor help doesn't work.

File: simulation_parameters/common/easter_egg_messages.json [hunk 5]
```diff
@@ -0,0 +1,35 @@
+{
+    "Messages": [
+        "Fun Fact, The Didinium  and Paramecium are a textbook example of a predator prey relationship that has been studied for decades, now are you the Didinium, or the Paramecium? Predator, or Prey?",
+        "Here's a tip, toxins can be used to knock other toxins away from you if you are quick enough.",
+        "Here's a tip, Osmoregulation costs 1 ATP per second per hex your cell has, each empty hex of cytoplasm generates 5 ATP per second aswell, which means if you are losing ATP due to osmoregulation just add a couple empty hexes cytoplasm or remove some organelles.",
+        "Fun Fact, In real life prokaryotes have something called Biocompartments which act like organelles, and are in fact called Polyhedral organelles",
+        "Fun Fact, The metabolosome is what is called a Polyhedral organelle",
+        "Here's a Tip, Chromatophores generate 1/3rd the glucose of a chloroplast",
+        "Here's a Tip, You generate exactly 2 glucose per second per chemoplast, as long as you have at least 1 hydrogen sulfide to convert.",
```
Comment (hhyyrylainen) [inline simulation_parameters/common/easter_egg_messages.json:9]: might be inaccurate

File: simulation_parameters/common/easter_egg_messages.json [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        "Here's a tip, sometimes its best just to run away from other cells.",
+        "Here's a tip, if a cell is about half your size, thats when you can engulf them.",
+        "Here's a tip, Bacteria can be stronger then they appear, they may look small, but some of them can burrow into you and kill you that way!",
+        "Here's a tip, You can hunt other species to extinction if you arent careful enough, they can also be hunted to extinction by other species.",
+        "Here's a tip, Every 5 minutes an Auto-evo step happens, if you dont evolve fast enough you may be out-competed.",
```
Comment (hhyyrylainen) [inline simulation_parameters/common/easter_egg_messages.json:14]: outdated

File: simulation_parameters/common/easter_egg_messages.json [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+        "Here's a tip, If you mouse over a cloud a box will pop up on the top left of your screen that tells you exactly whats there.",
```
Comment (hhyyrylainen) [inline simulation_parameters/common/easter_egg_messages.json:15]: outdated

File: simulation_parameters/common/easter_egg_messages.json (hunk 6 above)
Comment (Kasterisk) [reply simulation_parameters/common/easter_egg_messages.json:14]: removed

File: simulation_parameters/common/easter_egg_messages.json (hunk 7 above)
Comment (Kasterisk) [reply simulation_parameters/common/easter_egg_messages.json:15]: also removed

File: simulation_parameters/common/easter_egg_messages.json (hunk 5 above)
Comment (Kasterisk) [reply simulation_parameters/common/easter_egg_messages.json:9]: Should I remove those? (or add "these tips are inaccurate warnings" somewhere)

Comment (hhyyrylainen) [reply simulation_parameters/common/easter_egg_messages.json:9]: just remove them

**Comment by Kasterisk** [comment]
Okay, should be good now

**Pull request merged**